frb_codegen/src/generator/dart/mod.rs

```diff
@@ -221,48 +261,156 @@ fn generate_dart_declaration_body(
     )
 }
 
-fn generate_dart_implementation_body(
-    dart_funcs: &[GeneratedApiFunc],
-    dart_api2wire_funcs: &[String],
-    dart_api_fill_to_wire_funcs: &[String],
-    dart_wire2api_funcs: &[String],
-    dart_api_impl_class_name: &str,
-    dart_wire_class_name: &str,
-    dart_api_class_name: &str,
-) -> String {
-    format!(
-        "class {dart_api_impl_class_name} extends FlutterRustBridgeBase<{dart_wire_class_name}> implements {dart_api_class_name} {{
-            factory {dart_api_impl_class_name}(ffi.DynamicLibrary dylib) => {dart_api_impl_class_name}.raw({dart_wire_class_name}(dylib));
-
-            {dart_api_impl_class_name}.raw({dart_wire_class_name} inner) : super(inner);
-
-            {}
+fn section_header(header: &str) -> String {
+    format!("// Section: {}\n", header)
+}
 
-            // Section: api2wire
-            {}
+/// A Dart bridge module consists of several members:
+/// - An `_Impl` class exposing the public Rust functions
+/// - One or more `_Platform` classes implementing platform-specific helpers
+///
+/// The `_Impl` class takes a `_Platform _plat` instance as a private member,
+/// and the `_Platform` exposes all of its methods decorated as `@protected`.
```

Comment:
Nice doc :) Maybe we can move it to somewhere else such that more people who needs it can see it - currently it is on a non-pub function so I am worried, when someone wants to know high-level design, whether he will find this useful string

Maybe in the book? We have a chapter for internals: http://cjycode.com/flutter_rust_bridge/contributing/design.html